File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    iter_count, count_0 = 0, 0

    while s != '1':
        iter_count += 1
        count_0 += s.count('0')
        
        s = s.replace('0', '')
        c = len(s)
        s = bin(c)[2:]
        
    return [iter_count, count_0]

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.swift
import Foundation
/**
0. var x = s
1. xs의 모든 0 제거
2. String(len(s), radix: 2)
*/
func solution(_ s:String) -> [Int] {
    var x = s
    var deletedZeroCount = 0
    var result = 0
    
    while x != "1" {
        let deletedZero = x.filter { $0 != "0" }
        deletedZeroCount += x.count - deletedZero.count
        result += 1
        
        x = String(deletedZero.count, radix: 2)
    }
    
    return [result, deletedZeroCount]
}